testing/librsynctest.py

```diff
@@ -32,8 +32,12 @@ def sig_file_test_helper(self, blocksize, iterations, file_len = None):
 		for i in range(iterations):
 			MakeRandomFile(self.basis.path, file_len)
 			self._clean_file(self.sig)
-			assert not os.system(b"rdiff -b %i -H md4 signature %s %s" %
-								 (blocksize, self.basis.path, self.sig.path))
+			if b'-H' in subprocess.check_output(["rdiff", "--help"]):
+				assert not os.system("rdiff -b %i -H md4 signature %s %s" %
+									 (blocksize, self.basis.path, self.sig.path))
+			else:
+				assert not os.system("rdiff -b %i signature %s %s" %
```

Comment:
Needs to be bytes

Reply:
fixed the missing bytes in os.system call

Reply:
It's a trifle but replace %s with %b: b'%s' is deprecated, but will not be removed during the 3.x series.
See https://docs.python.org/3/library/stdtypes.html#printf-style-bytes-formatting

Reply:
didn't know the `b'%s'` deprecation, thanks for the hint, fixed in the touched lines